tests/test_version.py

```diff
@@ -0,0 +1,24 @@
+# (c) Copyright 2020 Hewlett Packard Enterprise Development LP
+
+# @author mandar shivrekar
+
+import pytest
+import os
+
+'''Version check test for NimOS package'''
+from tests.nimbleclientbase import SKIPTEST, log_to_file as log
+@pytest.mark.skipif(SKIPTEST is True,
+                    reason="skipped this test as SKIPTEST variable is true")
+def test_nimos_sdk_package_versioning():
+    log("**** Starting Tests for NimOS SDK package versioning TestCase *****\n")
+
+    version = None
+    base_path = os.path.dirname(__file__)
+    with open(os.path.join(base_path, '..', 'nimbleclient', '__init__.py')) as fh:
```

Comment:
this will create a problem and will fail on jenkins as nimbleclient folder will not be present at the same location where tests folder lies..

Ideally jenkins will just use pip install to install the sdk and then simply copy the test folder and run the test

the correct fix is as below

Reply:
@@ -3,7 +3,7 @@
 # @author mandar shivrekar

 import pytest
-import os
+from nimbleclient import __init__

 '''Version check test for NimOS package'''
 from tests.nimbleclientbase import SKIPTEST, log_to_file as log
@@ -13,8 +13,7 @@ def test_nimos_sdk_package_versioning():
     log("**** Starting Tests for NimOS SDK package versioning TestCase *****\n")

     version = None
-    base_path = os.path.dirname(__file__)
-    with open(os.path.join(base_path, '..', 'nimbleclient', '__init__.py')) as fh:
+    with open(__init__.__file__) as fh:
         info = fh.read()
         version = [line for line in info.split('\n') if line.startswith('__version__')][0].split(' = ')[1][1:-1]
         log("Obtained {!r} version for NimOS SDK package".format(version))

Reply:
note that we need to import init.py file with double underscores as forward and trailing space

`"+ from nimbleclient import __init__ "`

Reply:
Also, please make sure you do not have the nimbleclient folder present when you run the tests

Reply:
```
 import pytest
-import os
+from nimbleclient import __init__

 '''Version check test for NimOS package'''
 from tests.nimbleclientbase import SKIPTEST, log_to_file as log
@@ -13,8 +13,7 @@ def test_nimos_sdk_package_versioning():
     log("**** Starting Tests for NimOS SDK package versioning TestCase *****\n")

     version = None
-    base_path = os.path.dirname(__file__)
-    with open(os.path.join(base_path, '..', 'nimbleclient', '__init__.py')) as fh:
+    with open(__init__.__file__) as fh:
         info = fh.read()
         version = [line for line in info.split('\n') if line.startswith('__version__')][0].split(' = ')[1][1:-1]
         log("Obtained {!r} version for NimOS SDK package".format(version))

```